### PR TITLE
"Quick Fixes" to Assorted IDE Warnings

### PR DESCRIPTION
**Pull request by currantw**
Signed-off-by: currantw <taylor.curran@improving.com>

### Description

Cleanup. An assortment of "Quick Fixes" suggested by IntelliJ IDEA for warnings that it generates.

### Related Issues

Cleanup on files I happened to open as part of #3145.

### Check List
- N/A New functionality includes testing.
- N/A New functionality has been documented.
- N/A New functionality has javadoc added.
- N/A New functionality has a user manual doc added.
- N/A API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- N/A Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by YANG-DB** [comment]
@currantw pls resolve conflicts

File: core/src/main/java/org/opensearch/sql/analysis/Analyzer.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ public LogicalPlan visitRelation(Relation node, AnalysisContext context) {
 
   @Override
   public LogicalPlan visitRelationSubquery(RelationSubquery node, AnalysisContext context) {
-    LogicalPlan subquery = analyze(node.getChild().get(0), context);
+    LogicalPlan subquery = analyze(node.getChild().getFirst(), context);
```
Comment (YANG-DB) [inline core/src/main/java/org/opensearch/sql/analysis/Analyzer.java:177]: please create a method `visitFirstChild(node)` and replace all the `node.getChild().getFirst()` calls

**Review by YANG-DB — CHANGES_REQUESTED** [review]

**Comment by andy-k-improving** [comment]
I see there are different type of changes being apply in this PR( Ex: newer syntax of switch statement, getter substitution....etc) , should we separate these into multiple PR, under the same Github issue for the `code clean up` initiative? 
This will make review easily along also for better tracking.

**Comment by YANG-DB** [comment]
> I see there are different type of changes being apply in this PR( Ex: newer syntax of switch statement, getter substitution....etc) , should we separate these into multiple PR, under the same Github issue for the `code clean up` initiative? This will make review easily along also for better tracking.

I agree with @andy-k-improving lets split this into smaller chunks ...

**Comment by currantw** [comment]
> I see there are different type of changes being apply in this PR( Ex: newer syntax of switch statement, getter substitution....etc) , should we separate these into multiple PR, under the same Github issue for the `code clean up` initiative? This will make review easily along also for better tracking.

I'm not sure what the established process is for this (or if there is one), but my fear is that this would create a lot of overhead, and probably discourage developers (myself included) from taking on any "ad hoc" cleanup tasks if we need to create a new branch and new PR for each "type" of cleanup change. 

These are "Quick Fixes" suggested by IntelliJ to warnings that the IDE itself generates, so they are _intended_ to be uncontroversial changes. Moreover, these are just files that I happened to open (and notice these warnings) while working on other issues (rather than a comprehensive effort to fix one or more particular types of warning). My general thought is that we should be encouraging developers to do minor cleanup "as they go", rather than waiting for "big bang" dedicated cleanup work -- which tends to get done infrequently, if at all...

Will post on Slack to see if anyone else on the team has thoughts!

**Comment by currantw** [comment]
> > I see there are different type of changes being apply in this PR( Ex: newer syntax of switch statement, getter substitution....etc) , should we separate these into multiple PR, under the same Github issue for the `code clean up` initiative? This will make review easily along also for better tracking.
> 
> I agree with @andy-k-improving lets split this into smaller chunks ...

Okay, sounds good. I will close and create new separate PRs for each issue.

**Pull request closed without merging**